include/reactor-cpp/multiport_callback.hh

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2022 TU Dresden
+ * All rights reserved.
+ *
+ * Authors:
+ *   Tassilo Tanneberger
+ */
+
+#ifndef REACTOR_CPP_MULTIPORT_CALLBACK_HH
+#define REACTOR_CPP_MULTIPORT_CALLBACK_HH
+
+#include <algorithm>
+#include <atomic>
+#include <iostream>
+#include <type_traits>
+#include <vector>
+
+namespace multiport {
+
+// fancy custom type_trait taken from stackoverflow
+// which checks if given class has the member function deactivate
+template <typename T> class has_deactivate {
+  using one = char;
+  struct two {
+    char x[2]; // NOLINT avoid-c-arrays
+  };
+
+  template <typename C> static auto test(decltype(&C::has_deactivate)) -> one;
+  template <typename C> static auto test(...) -> two;
+
+public:
+  enum { value = sizeof(test<T>(0)) == sizeof(char) };
+};
+
+// struct which gets handed to the ports to they can talk back
+// to the portbank
+struct LockedPortList {
+  std::atomic<std::size_t>* size_ = nullptr;
+  std::vector<std::size_t>* active_ports_ = nullptr;
+};
+
+template <class T, class A = std::allocator<T>>
+class PortBankCallBack { // NOLINT cppcoreguidelines-special-member-functions
```

Comment:
I am confused by this name.
1. There is no PortBank in LF. The right name would be Multiport.
2. Is this a callback or a multiport? It looks like it is a Multiport which _has_ a callback. So we could just name this class Multiport, right?